azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -201,15 +215,44 @@ private boolean isMultiExecutorMode() {
    */
   private void refreshExecutors() {
     synchronized (activeExecutors) {
-      // TODO: rest api call to refresh executor stats
+      ExecutorService taskExecutors =
+        Executors.newFixedThreadPool(activeExecutors.size());
+      for (final Executor executor : activeExecutors) {
+
+        // execute each executorInfo refresh task
+        taskExecutors.execute(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              String jsonResponse =
+                callExecutorForJsonString(executor.getHost(),
+                  executor.getPort(), "/serverstastics", null);
+              executor.setExecutorInfo(ExecutorInfo
+                .fromJSONString(jsonResponse));
+            } catch (Exception e) {
+              logger.error("Failed to update ExecutorInfo executorId :"
+                + executor, e);
+            }
+          }
+        });
+      }
+
+      taskExecutors.shutdown();
+      try {
+        // wait 5 seconds for all executors to be refreshed
+        taskExecutors.awaitTermination(5, TimeUnit.SECONDS);
+      } catch (InterruptedException e) {
+        logger.error("Timed out while waiting for executorInfo refresh", e);
```

Comment:
What are the impacts when tasks don't come back in 5 seconds?  stale statistics?

It would be nice to have visibility into this refresh tasks  - i.e last refresh timestamp in the JMX page.